File: snippets/fsharp/System/IndexOutOfRangeException/Overview/foreach1.fs
module foreach1

// <Snippet7>
open System

let populateArray items maxValue =
    let rnd = Random()
    [| for i = 0 to items - 1 do
        rnd.Next(0, maxValue + 1) |]

// Generate array of random values.
let values = populateArray 5 10
// Display each element in the array.
for value in values do
    printf $"{values[value]}   "

// The example displays output like the following:
//    6   4   4
//    Unhandled Exception: System.IndexOutOfRangeException:
//    Index was outside the bounds of the array.
//       at <StartupCode$fs>.main@()
// </Snippet7>

File: snippets/csharp/System/IndexOutOfRangeException/Overview/foreach1.cs
// <Snippet7>
using System;

public class ForeachExample1
{
    public static void Run()
    {
        // Generate array of random values.
        int[] values = PopulateArray(5, 10);
        // Display each element in the array.
        foreach (int value in values)
            Console.Write($"{values[value]}   ");
    }

    private static int[] PopulateArray(int items, int maxValue)
    {
        int[] values = new int[items];
        Random rnd = new();
        for (int ctr = 0; ctr < items; ctr++)
            values[ctr] = rnd.Next(0, maxValue + 1);

        return values;
    }
}
// The example displays output like the following:
//    6   4   4
//    Unhandled Exception: System.IndexOutOfRangeException:
//    Index was outside the bounds of the array.
//       at Example.Main()
// </Snippet7>
